### ISSUE TITLE
Do not create .stignore if there are no ignore patterns

### ISSUE THREAD

**Issue by Otiel** [syncthing/syncthing#6190]
### TLDR;

Syncthing should not create empty `.stignore` files as it clutters the folder contents.

### Prerequisites

1. Choose a folder that doesn't have any Ignore patterns.
2. Make sure there's no existing `.stignore` file or delete it.

### Steps to reproduce

1. Go to the Syncthing GUI, and click "Edit" the selected folder.
2. Hit "Save" (without changing anything).

### Actual results

A new, empty `.stignore` file is created.

### Expected results

When hitting "Save", if there are no Ignore patterns specified, Syncthing should not create a `.stignore` file, or should delete it if it already exists.

### Version Information

Syncthing Version: v1.3.1

**Comment by AudriusButkevicius** [comment]
The file is hidden, whats the problem?

**Comment by calmh** [comment]
I agree that an empty .stignore is pointless and not having at all is cleaner. Minor issue for someone interested to take on.

**Comment by Otiel** [comment]
> The file is hidden, why is that a problem?

1. It's not hidden for every configuration (for instance, accessing a Linux file system through a Windows file explorer).
2. There are some people that choose to show hidden files (myself included), so for those the clutter is visible.
3. As said by @calmh, it is cleaner to not have an empty unused file hanging somewhere in the user's files.

**Comment by AudriusButkevicius** [comment]
There is a counter argument for things like that encfs issue raised recently where the filesystem returns something funny when checked for the existence of the file, so it might be better to create and keep and empty file than fail overall. 

I guess people could just put a comment in the file.

**Comment by PNMarano** [comment]
Perhaps a dumb question: What about just moving everything Syncthing related (including stuff like .stversions) into one hidden folder (maybe .stfolder?)? We'd eliminate the clutter, while still allowing options for expanding features later on.

**Comment by calmh** [comment]
That's indeed the raison d'être of the .stfolder, and https://github.com/syncthing/syncthing/issues/6001.
